Source/DafnyDriver/Commands/DafnyCli.cs

```diff
@@ -489,11 +488,18 @@ class WritersConsole : IConsole {
     // which is not handled well.
     if (options.Get(CommonOptionBag.UseStandardLibraries)) {
       var reporter = new ConsoleErrorReporter(options);
+      if (options.CompilerName is null or "cs" or "java" or "go" or "py" or "js") {
```

Comment:
Could you introduce a boolean name indicating why you perform this test? e.g. like "noTargetNeeded".

Reply:
This is from my PR (#4790), it's just moving in this one. I intend to clean this up together with some other stdlibs follow-up - I figure it would be better to add an optional `StdLibDooFile` property on backends instead.

Reply:
I copied this from DafnyCLI, but it checks whether a target specific .doo file exists. The block won't be necessary anymore once we merge the replaceable PR.